[Metron-352] Updated Deployment to use HDP 2.4

jjmeyer0 opened the pull request:
Change the deployment to use the Ambari 2.2.2.0 and HDP 2.4.2.0 repositories. This included consolidating some Ansible variables into group_vars and fixing typos.

jjmeyer0 commented:
Hello, has anyone had issues with org.apache.metron.enrichment.integration.EnrichmentIntegrationTest hanging? It seems to be trhowing the same exception over and over again which cause Travis CI to kill the build. I've looked at builds that were successful. The ones I've checked also have this error, just not this many times. Also, I'm having trouble reproducing it locally. Has anyone seen this before?

dlyle65535 commented:
Hi JJ,

Thanks for taking this on. Looks great! One small thing, could you look at moving the repeated group vars (e.g. hdp_stack, hdp_repo_def and the ambari variables) to defaults in the appropriate roles?

Also, since you are in the neighborhood: METRON-43 could be satisfied by removing the passwd_less_ssh role. Since it's no longer needed anyway, rather than update it to reflect these changes, how would you feel about a twofer?

jjmeyer0 commented:
Hello David,

Thanks for the feedback. I moved the group_vars to defaults. Also, I squashed my commits to make it easier to read. I also deleted the passwd_less_ssh role as well.

dlyle65535 commented:
Outstanding. Great work! How did you test this, full_dev on Vagrant?

jjmeyer0 commented:
Thanks, I appreciate it. Yes, I tested this with full_dev on Vagrant. I followed this [documentation](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds).

dlyle65535 commented:
Great, I'm +1. Thanks for this.

nickwallen commented:
+1 Very nice.  A lot of good things cleaned-up here.
